### PR TITLE
fix(client): Ensure tsc can create a declaration out of extended client

### LINKED ISSUE #19866: 4.16.x cannot wrap `$extend` in factory function when `compilerOptions.composite` is `true`
### Bug description

I'm trying to upgrade to 4.16.x and use the GA version of client extensions. However, now when I try to export my extended client from a file I get the following typescript error.

> Return type of exported function has or is using name 'PrismaPromise_2' from external module "node_modules/.prisma/client/runtime/index" but cannot be named. (tsserver 4023)

### How to reproduce

1. clone https://github.com/awinograd/prisma-repro/tree/aw/extend
1. `yarn tsc --build --force`

```
✘ ⮀ prisma-repro ⮀ ⭠ aw/extend± ⮀ yarn tsc --build --force
yarn run v1.22.15
$ /Users/awinograd/programming/prisma-repro/node_modules/.bin/tsc --build --force
script.ts:3:17 - error TS4058: Return type of exported function has or is using name 'PrismaPromise_2' from external module "/Users/awinograd/programming/prisma-repro/node_modules/@prisma/client/runtime/index" but cannot be named.

3 export function getDbClient(options: Prisma.PrismaClientOptions) {
                  ~~~~~~~~~~~

script.ts:3:17 - error TS7056: The inferred type of this node exceeds the maximum length the compiler will serialize. An explicit type annotation is needed.

3 export function getDbClient(options: Prisma.PrismaClientOptions) {
                  ~~~~~~~~~~~


Found 2 errors.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Expected behavior

extended client can be exported without a TS error

### Environment & setup

- OS: macOS
- Database:  PostgreSQL
- Node.js version: 16.13.1

### Prisma Version

```
prisma                  : 4.16.0
@prisma/client          : 4.16.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine b20ead4d3ab9e78ac112966e242ded703f4a052c (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli b20ead4d3ab9e78ac112966e242ded703f4a052c (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.16.0-66.b20ead4d3ab9e78ac112966e242ded703f4a052c
Default Engines Hash    : b20ead4d3ab9e78ac112966e242ded703f4a052c
Studio                  : 0.484.0
```

### PR DESCRIPTION
**Pull request by SevInf**
Inside of runtime we use nested namespaces for some of the helper types
(`Types.Extensions.ExtendsHook`). When creating a declaration from the
users code, `tsc` can't trace them back to `runtime` module, attempts to
inline them instead and fails.

Fixes the problem by exporting offending types from the root of the
runtime module rather then nested namespace. This is enough for `tsc` to
reference type by the name rather than attempt to inline it.

Fix #19866

### CODE REVIEW

**Review by Jolg42 — APPROVED** [review]

File: packages/client/tests/e2e/issues/19866-ts-composite-declaration/readme.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Readme
+
+This is an example of a readme file. Put a description of your test here.
```
Comment (aqrln) [inline packages/client/tests/e2e/issues/19866-ts-composite-declaration/readme.md:3]:
```suggestion
# 19866-ts-composite-declaration

Regression test for <https://github.com/prisma/prisma/issues/19866>.
```

**Review by aqrln — APPROVED** [review]
LGTM, suggestions are non-blocking

**Review by jkomyno — APPROVED** [review]
Nice catch!

File: packages/client/src/runtime/index.ts [hunk 2]
```diff
@@ -38,10 +39,13 @@ export { Public }
...
+ * Payload, PrismaPromise and Extensions types are already exported via Types but tsc
+ * won't be able to trace them correctly back to runtime module and fail with either
+ * "... is using the type X but can not be named" or "The inferred type of this node exceeds the maximum length" error.
  * The issue lies with the type bundler which does not add exports for dependent types
  * TODO: Maybe simply exporting all types in runtime will do the trick
  */
-export { type Payload }
+export { type Payload, type PrismaPromise }
 
+export * from './core/types/Extensions'
```
[the remark this answers is not in the document]
Comment (SevInf) [reply packages/client/src/runtime/index.ts:50]: I tried, but prettier objected

**Pull request merged**